client/rpc/block.go

```diff
@@ -0,0 +1,63 @@
+package rpc
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	flagSelect = "select"
+)
+
+func blockCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "block [height]",
+		Short: "Get verified data for a the block at given height",
+		RunE:  getBlock,
+	}
+	cmd.Flags().StringP(client.FlagNode, "n", "tcp://localhost:46657", "Node to connect to")
+	// TODO: change this to false when we can
+	cmd.Flags().Bool(client.FlagTrustNode, true, "Don't verify proofs for responses")
+	cmd.Flags().StringSlice(flagSelect, []string{"header", "tx"}, "Fields to return (header|txs|results)")
+	return cmd
+}
+
+func getBlock(cmd *cobra.Command, args []string) error {
+	var height *int64
+	// optional height
+	if len(args) > 0 {
+		h, err := strconv.Atoi(args[0])
+		if err != nil {
+			return err
+		}
+		if h > 0 {
+			tmp := int64(h)
+			height = &tmp
+		}
+	}
+
+	// get the node
+	uri := viper.GetString(client.FlagNode)
+	node := client.GetNode(uri)
+
+	// TODO: actually honor the --select flag!
+	// header -> BlockchainInfo
+	// header, tx -> Block
+	// results -> BlockResults
+	res, err := node.Block(height)
+	if err != nil {
+		return err
+	}
+
+	output, err := json.MarshalIndent(res, "  ", "")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(output))
```

Comment:
the output of this is pretty gross. any idea what's going on ?

Reply:
go-wire nightmare

Reply:
You mean the signature as an array of ints... same as status with the pubkey?
I can make it use go-wire json.... a bit nicer but still not

Reply:
Before:
```
./build/basecli status
{
  "node_info": {
  "pub_key": [
  18,
  24,
  175,
  75,
  128,
  241,
  196,
  70,
  192,
  62,
  146,
  211,
  201,
  156,
  153,
  122,
  69,
  169,
  116,
  12,
  206,
  247,
  243,
  151,
  8,
  229,
  60,
  131,
  144,
  226,
  127,
  157
  ],
  "listen_addr": "10.1.23.73:46656",
  "network": "test-chain-Dod558",
  "version": "0.16.0",
  "channels": "QCAhIiMwOAA=",
  "moniker": "micropower",
  "other": [
  "wire_version=0.7.2",
  "p2p_version=0.5.0",
  "consensus_version=v1/0.2.2",
  "rpc_version=0.7.0/3",
  "tx_index=on",
  "rpc_addr=tcp://0.0.0.0:46657"
  ]
  },
  "pub_key": [
  83,
  54,
  32,
  163,
  112,
  137,
  3,
  178,
  100,
  246,
  90,
  29,
  153,
  69,
  58,
  100,
  60,
  168,
  173,
  185,
  98,
  190,
  7,
  136,
  191,
  183,
  16,
  224,
  39,
  200,
  219,
  216
  ],
  "latest_block_hash": "F09AFFF7CBE65794A46528643D7A654A9FA10756",
  "latest_app_hash": "F38F97BCD4E37C09FBBBFEAF86B3770B4F00A967",
  "latest_block_height": 15,
  "latest_block_time": "2018-02-28T15:32:20.755406735+01:00",
  "syncing": false
  }
```

Reply:
After:
```
./build/basecli status
{"node_info":{"pub_key":{"_df":"B3EF72328EAF59","_v":"EhivS4DxxEbAPpLTyZyZekWpdAzO9/OXCOU8g5Dif50="},"listen_addr":"10.1.23.73:46656","network":"test-chain-Dod558","version":"0.16.0","channels":"QCAhIiMwOAA=","moniker":"micropower","other":["wire_version=0.7.2","p2p_version=0.5.0","consensus_version=v1/0.2.2","rpc_version=0.7.0/3","tx_index=on","rpc_addr=tcp://0.0.0.0:46657"]},"pub_key":{"_df":"B3EF72328EAF59","_v":"UzYgo3CJA7Jk9lodmUU6ZDyorblivgeIv7cQ4CfI29g="},"latest_block_hash":"66760BB85DB635978C33601DF5E458816B03A8C5","latest_app_hash":"0BCDF9F6F56D31F9D0806BCFE7204EB5E3B98AF9","latest_block_height":206,"latest_block_time":"2018-02-28T15:35:38.56783532+01:00","syncing":false}
```